proposed/links.md

```diff
@@ -162,14 +181,92 @@ interface LinkInterface
 }
 ~~~
 
-#### 3.2.1 `Psr\Http\Link\LinkCollectionInterface`
+### 3.2 `Psr\Http\Link\EvolvableLinkInterface`
+
+~~~php
+<?php
+
+namespace Psr\Http\Link;
+
+/**
+ * An evolvable link value object.
+ */
+interface EvolvableLinkInterface extends LinkInterface
+{
+    /**
+     * Returns an instance with the specified href.
+     *
+     * @param string $href
+     *   The href value to include.  It must be one of:
+     *     - An absolute URI, as defined by RFC 5988.
+     *     - A relative URI, as defined by RFC 5988. The base of the relative link
+     *       is assumed to be known based on context by the client.
+     *     - A URI template as defined by RFC 6570.
+     *     - An object implementing __toString() that produces one of the above
+     *       values.
+     *
+     * An implementing library SHOULD evaluate a passed object to a string
+     * immediately rather than waiting for it to be returned later.
+     *
+     * @return self
```

Comment:
`s/self/static/`. `self` implies the same instance is returned, while `static` implies a new instance. (This is something clarified after PSR-7 was ratified.)